src/game_player.cpp

```diff
@@ -218,8 +258,21 @@ void Game_Player::UpdateSelfMovement() {
 			}
 		}
 
-		// ESC-Menu calling
-		if (Game_System::GetAllowMenu() && !Game_Message::message_waiting && !IsMoveRouteOverwritten() && Input::IsTriggered(Input::CANCEL)) {
+		if (IsStopping()) {
+			if (Input::IsTriggered(Input::DECISION)) {
+				if (!GetOnOffVehicle()) {
+					CheckActionEvent();
+				}
+			}
+		}
+	}
+
+	// ESC-Menu calling
+	if (Game_System::GetAllowMenu()
+			&& !Game_Message::visible
+			&& !Game_Map::GetInterpreter().IsRunning())
+	{
```

Comment:
> Refactor player update routine order 

I'm not sure about the change for ESC-Menu calling.

I thought the menu can't be accessed when the route is overwritten or a message is pending to be shown but both these conditions were removed.

Reply:
`UpdateSelfMovement()` is not called by `Game_Character` if `IsMoveRouteOverwritten() == true`. 

For the message pending part, I'll need to check how that overlaps with visible.

Reply:
It's confusing as we don't have a standard way to ask "Is there a message active?"

`message_waiting` goes from when the `ShowMessage` command was invoked up until the message box starts to close. `visible` goes from the first frame of the actual message until the message close animation is done. So in most cases you probably want both.

I changed this to check for both. Also I added back a check for `!IsMoveRouteOverwritten()` I guess in theory if you press a direction and ESC at the same time, the player directional move could cause an event to run, which then sets a move route, which then prevents the player from opening the menu in the same frame...